editor/plugins/shader_baker_export_plugin.cpp

```diff
@@ -0,0 +1,461 @@
+	if (!_is_active(p_features)) {
+		return false;
+	}
+
+	if (!_initialize_container_format(p_platform, p_features)) {
+		return false;
+	}
+
+	shader_cache_platform_name = p_platform->get_os_name();
+	shader_cache_renderer_name = RendererSceneRenderRD::get_singleton()->get_name();
+	tasks_processed = 0;
+	tasks_total = 0;
+	tasks_cancelled = false;
+
+	StringBuilder to_hash;
+	to_hash.append("[GodotVersionNumber]");
+	to_hash.append(VERSION_NUMBER);
+	to_hash.append("[GodotVersionHash]");
+	to_hash.append(VERSION_HASH);
+	to_hash.append("[Renderer]");
+	to_hash.append(shader_cache_renderer_name);
+	customization_configuration_hash = to_hash.as_string().hash64();
+
+	BitField<RenderingShaderLibrary::FeatureBits> renderer_features = {};
+	bool xr_enabled = GLOBAL_GET("xr/shaders/enabled");
+	renderer_features.set_flag(RenderingShaderLibrary::FEATURE_ADVANCED_BIT);
+	if (xr_enabled) {
+		renderer_features.set_flag(RenderingShaderLibrary::FEATURE_MULTIVIEW_BIT);
+	}
+
+	int vrs_mode = GLOBAL_GET("rendering/vrs/mode");
+	if (vrs_mode != 0) {
+		renderer_features.set_flag(RenderingShaderLibrary::FEATURE_VRS_BIT);
+	}
+
+	RendererSceneRenderRD::get_singleton()->enable_features(renderer_features);
+
+	// Included all shaders created by renderers and effects.
+	ShaderRD::shaders_embedded_set_lock();
+	const ShaderRD::ShaderVersionPairSet &pair_set = ShaderRD::shaders_embedded_set_get();
+	for (Pair<ShaderRD *, RID> pair : pair_set) {
+		_customize_shader_version(pair.first, pair.second);
+	}
+
+	ShaderRD::shaders_embedded_set_unlock();
+
+	// Include all shaders created by embedded materials.
+	RendererRD::MaterialStorage *material_storage = RendererRD::MaterialStorage::get_singleton();
+	material_storage->shader_embedded_set_lock();
+	const HashSet<RID> &rid_set = material_storage->shader_embedded_set_get();
+	for (RID rid : rid_set) {
+		RendererRD::MaterialStorage::ShaderData *shader_data = material_storage->shader_get_data(rid);
+		if (shader_data != nullptr) {
+			Pair<ShaderRD *, RID> shader_version_pair = shader_data->get_native_shader_and_version();
+			if (shader_version_pair.first != nullptr) {
+				_customize_shader_version(shader_version_pair.first, shader_version_pair.second);
+			}
+		}
+	}
+
+	material_storage->shader_embedded_set_unlock();
+
+	return true;
+}
+
+bool ShaderBakerExportPlugin::_begin_customize_scenes(const Ref<EditorExportPlatform> &p_platform, const Vector<String> &p_features) {
+	if (!_is_active(p_features)) {
+		return false;
+	}
+
+	if (shader_container_format == nullptr) {
+		// Resource customization failed to initialize.
+		return false;
+	}
+
+	return true;
+}
+
+void ShaderBakerExportPlugin::_end_customize_resources() {
+	if (!_initialize_cache_directory()) {
+		return;
+	}
+
+	// Run a progress bar that waits for all shader baking tasks to finish.
+	bool progress_active = true;
+	EditorProgress editor_progress("baking_shaders", TTR("Baking shaders"), tasks_total);
+	editor_progress.step("Baking...");
```

Comment:
Cosmetics: add ",0)" so the first step does not start with -1.

Reply:
Added, thanks!